browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -128,6 +128,8 @@ BraveBrowserView::BraveBrowserView(std::unique_ptr<Browser> browser)
       kTabsSearchShow,
       base::BindRepeating(&BraveBrowserView::OnPreferenceChanged,
                           base::Unretained(this)));
+  // Show the correct value in settings on initial start
+  BraveBrowserView::OnPreferenceChanged(kTabsSearchShow);
```

Comment:
How about creating `UpdateSearchTabButtonState()` and move `OnPreferenceChanged`'s impl to it?
Then, calling `UpdateSearchTabButtonState()` from here and from `OnPreferenceChanged()` ?